File: skvdb-client/src/main/java/org/skvdb/network/ConnectionFactory.java
package org.skvdb.network;

public class ConnectionFactory {
    private String host;

    private int port;

    private String username;

    private String password;

    public Connection createConnection() {
        return new Connection(host, port, username, password);
    }

    public void setHost(String host) {
        this.host = host;
    }

    public void setPort(int port) {
        this.port = port;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public void setPassword(String password) {
        this.password = password;
    }
}

File: go-skvdb-client/src/conn/connectionFactory.go
package conn

import (
	"fmt"
)

type ConnectionFactory struct {
	host     string
	port     int
	username string
	password string
}

func NewConnectionFactory(host string, port int, username string, password string) *ConnectionFactory {
	return &ConnectionFactory{host: host, port: port, username: username, password: password}
}

func (factory *ConnectionFactory) CreateConnection() (*Connection, error) {
	const op = "network.ConnectionFactory.CreateConnection"
	connection, err := newConnection(factory.host, factory.port, factory.username, factory.password)
	if err != nil {
		return nil, fmt.Errorf("%s: %w", op, err)
	}
	return connection, nil
}
